pyphare/pyphare/pharein/__init__.py

```diff
@@ -251,9 +251,14 @@ def as_paths(rb):
     addInitFunction(maginit_path + "y_component", fn_wrapper(modelDict["by"]))
     addInitFunction(maginit_path + "z_component", fn_wrapper(modelDict["bz"]))
 
+    serialized_sim = serialize_sim(simulation)
+
     #### adding diagnostics
+
     diag_path = "simulation/diagnostics/"
     for diag in list(simulation.diagnostics.values()):
+        diag.attributes["serialized_simulation"] = serialized_sim
```

Comment:
this doesn't have to be like this exactly, but this does work, and is convenient.